index.js

```diff
@@ -143,18 +69,29 @@ module.exports = function (newrelic, opts) {
 	} else {
 		// newrelic has frontend display logic, which will format the transaction name if it's under express
 		// (method, path) => 'Expressjs/' + method + '/' + path;
-		parseTransactionName = (method, path) => 'Koajs/' + (path[0] === '/' ? path.slice(1) : path) + '#' + method;
+		parseTransactionName = DEFAULT_CUSTOME_TRANSACTION_NAME;
 	}
 
 	function setTransactionName(method, path) {
 		newrelic.setTransactionName(parseTransactionName(method, path));
 	}
 
 	return async function koaNewrelic(ctx, next) {
-		if (opts.usePathAsDefaultRoute) {
-			setTransactionName(ctx.method, ctx.path);
-		}
-
+		// for patching the render method
+		Object.defineProperty(ctx, renderMethodName, {
```

Comment:
this can even be generalised for every function under `ctx`, receive a list of functions
check if it's under `ctx` be default, and do following, then wrap the the _original by set